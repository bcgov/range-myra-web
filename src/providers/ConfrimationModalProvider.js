import { Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
import React, { createContext, useCallback, useContext, useRef, useState } from 'react';
import { PrimaryButton } from '../components/common';
const ConfirmationModal = createContext();

export function ConfirmationModalProvider({ children }) {
  const [state, setState] = useState({ isOpen: false });
  const fn = useRef();

  const confirm = useCallback(
    (data) => {
      return new Promise((resolve) => {
        setState({ ...data, isOpen: true });
        fn.current = (choice) => {
          resolve(choice);
          setState({ isOpen: false });
        };
      });
    },
    [setState],
  );

  return (
    <ConfirmationModal.Provider value={confirm}>
      {children}
      <Dialog
        open={state.isOpen}
        onClose={() => fn.current(false)}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">{state.titleText || 'Confirm'}</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">{state.contentText}</DialogContentText>
        </DialogContent>
        <DialogActions>
          <PrimaryButton inverted onClick={() => fn.current(false)}>
            {state.cancelText || 'Cancel'}
          </PrimaryButton>
          <PrimaryButton onClick={() => fn.current(true)} autoFocus>
            {state.confirmText || 'Confirm'}
          </PrimaryButton>
        </DialogActions>
      </Dialog>
    </ConfirmationModal.Provider>
  );
}

export default function useConfirm() {
  return useContext(ConfirmationModal);
}
