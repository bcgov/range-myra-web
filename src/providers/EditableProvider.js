import React, { useContext } from 'react'
import PropTypes from 'prop-types'

export const EditableContext = React.createContext({})

export const useEditable = () => useContext(EditableContext)

const EditableProvider = ({ children, editable }) => {
  return (
    <EditableContext.Provider value={editable}>
      {children}
    </EditableContext.Provider>
  )
}

EditableProvider.propTypes = {
  children: PropTypes.node
}

export default EditableProvider
