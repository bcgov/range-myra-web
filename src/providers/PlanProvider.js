import React, { useContext, useState, useEffect } from 'react'
import { normalize } from 'normalizr'
import * as API from '../api'
import { storePlan } from '../actions'
import * as schema from '../actionCreators/schema'
// import * as schema from '../schema'
import { getNetworkStatus } from '../utils/helper/network'
import { connect } from 'react-redux'
import { appendUsage } from '../utils'

/**
 * @typedef {Object} PlanContext
 * @property {string|number} currentPlanId ID of the current plan
 * @property {object|null} currentPlan Current plan
 * @property {boolean} isFetchingPlan Is the plan currently being fetched
 * @property {boolean} isSavingPlan Is the plan saving
 * @property {string|null} errorFetchingPlan Defined if there was an error fetching the plan
 * @property {(id?: number) => Promise<object>} fetchPlan Fetches by default the plan with id `currentPlanId`. Returns the plan, as well as sets `currentPlan`. If no network connectivity, fallbacks to local storage.
 * @property {(plan: object) => Promise<number>} savePlan Saves `plan` to either the remote backend or local storage, depending on network connectivity.
 */

/**
 * @type {React.Context<PlanContext>}
 */
const PlanContext = React.createContext()

/**
 * @returns {PlanContext}
 */
export const useCurrentPlan = () => useContext(PlanContext)

export const PlanProvider = ({ children, storePlan }) => {
  const [currentPlanId, setCurrentPlanId] = useState(null)
  const [currentPlan, setCurrentPlan] = useState(null)
  const [isFetchingPlan, setFetchingPlan] = useState(false)
  const [isSavingPlan, setSavingPlan] = useState(false)
  const [errorFetchingPlan, setErrorFetchingPlan] = useState(null)

  const fetchPlan = async (planId = currentPlanId) => {
    setFetchingPlan(true)

    try {
      const plan = await API.getPlan(planId)
      setCurrentPlan(appendUsage(plan))

      // TODO: remove redux
      const isOnline = await getNetworkStatus()
      if (isOnline) {
        // fetchRUP(planId)
        storePlan(normalize(plan, schema.plan))
      }

      return plan
    } catch (e) {
      setErrorFetchingPlan(e)
    } finally {
      setFetchingPlan(false)
    }
  }

  const savePlan = async plan => {
    setSavingPlan(true)

    const planId = await API.savePlan(plan)
    await fetchPlan(planId)

    setSavingPlan(false)

    return planId
  }

  useEffect(() => {
    if (currentPlanId !== null) {
      fetchPlan()
    }
  }, [currentPlanId])

  return (
    <PlanContext.Provider
      value={{
        setCurrentPlanId,
        currentPlan,
        isFetchingPlan,
        isSavingPlan,
        errorFetchingPlan,
        fetchPlan,
        savePlan
      }}>
      {children}
    </PlanContext.Provider>
  )
}

export default connect(
  null,
  { storePlan }
)(PlanProvider)
