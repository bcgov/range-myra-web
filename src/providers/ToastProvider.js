import React, { useContext, useState, useRef, useEffect } from 'react';
import uuid from 'uuid-v4';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { Icon } from 'semantic-ui-react';

export const ToastContext = React.createContext({});

export const useToast = () => useContext(ToastContext);

const ToastProvider = ({ children }) => {
  const [toasts, setToasts] = useState([]);
  const toastsRef = useRef();

  useEffect(() => {
    toastsRef.current = toasts;
  }, [toasts]);

  const addToast = (message, status, config = {}) => {
    const { timeout = 3000, content } = config;
    const id = uuid();

    setToasts([
      ...toastsRef.current,
      {
        id,
        message,
        status,
        content,
      },
    ]);

    setTimeout(() => {
      removeToast(id);
    }, timeout);

    return id;
  };

  const successToast = (message, config) => addToast(message, 'success', config);
  const errorToast = (message, config) => addToast(message, 'error', config);
  const warningToast = (message, config) => addToast(message, 'warning', config);

  const removeToast = (id) => setToasts(toastsRef.current.filter((t) => t.id !== id));

  return (
    <ToastContext.Provider
      value={{
        addToast,
        successToast,
        errorToast,
        warningToast,
        removeToast,
      }}
    >
      <section className="toasts">
        {toasts.map((toast) => (
          <div key={toast.id} className="toast">
            <div
              className={classnames('toast__icon', {
                toast__icon__success: toast.status === 'success',
                toast__icon__error: toast.status === 'error',
                toast__icon__warning: toast.status === 'warning',
              })}
            >
              {toast.status === 'success' && <Icon name="check circle" size="large" />}
              {toast.status === 'error' && <Icon name="warning circle" size="large" />}
              {toast.status === 'warning' && <Icon name="warning" size="large" />}
            </div>
            <div className="toast__content">
              {toast.message}
              {toast.content}
            </div>
            <button className="toast__dismiss" onClick={() => removeToast(toast.id)}>
              <Icon name="times" size="small" />
            </button>
          </div>
        ))}
      </section>
      {children}
    </ToastContext.Provider>
  );
};

ToastProvider.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired,
    givenName: PropTypes.string.isRequired,
    familyName: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    roles: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  }),
  children: PropTypes.node,
};

export default ToastProvider;
