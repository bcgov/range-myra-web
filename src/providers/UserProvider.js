import React, { useContext } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getUser } from '../reducers/rootReducer';

export const UserContext = React.createContext({});

export const useUser = () => useContext(UserContext);

const UserProvider = ({ user, children }) => {
  return <UserContext.Provider value={user}>{children}</UserContext.Provider>;
};

UserProvider.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired,
    givenName: PropTypes.string.isRequired,
    familyName: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    roles: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  }),
  children: PropTypes.node,
};

// Just take user from redux store and make it accessible via UserContext for now
// Eventually handle retrieving and storing the user in this component w/o redux
const mapStateToProps = (state) => ({
  user: getUser(state),
});

export default connect(mapStateToProps)(UserProvider);
