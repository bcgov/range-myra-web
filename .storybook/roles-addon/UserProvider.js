import React, { useState, useEffect } from 'react';
import { UserContext } from '../../src/providers/UserProvider';
import user from '../mocks/user';
import addons from '@storybook/addons';

const UserProvider = ({ children }) => {
  const [currentRole, setCurrentRole] = useState('myra_range_officer');

  const channel = addons.getChannel();

  const handleChange = (role) => {
    setCurrentRole(role);
  };

  useEffect(() => {
    channel.on('role/change', handleChange);

    return () => {
      channel.removeListener('role/change', handleChange);
    };
  }, []);

  return (
    <UserContext.Provider
      value={{
        ...user,
        roles: [currentRole],
      }}
    >
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
