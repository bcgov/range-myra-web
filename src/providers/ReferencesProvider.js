import React, { useContext, useEffect } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { getReferences } from '../reducers/rootReducer'
import { fetchReferences } from '../actionCreators'
import { useUser } from './UserProvider'

export const ReferencesContext = React.createContext({})

export const useReferences = () => useContext(ReferencesContext)

const ReferencesProvider = ({ references, fetchReferences, children }) => {
  const user = useUser()

  useEffect(() => {
    fetchReferences()
  }, [user])

  return (
    <ReferencesContext.Provider value={references}>
      {children}
    </ReferencesContext.Provider>
  )
}

ReferencesProvider.propTypes = {
  references: PropTypes.shape({}).isRequired,
  fetchReferences: PropTypes.func.isRequired,
  children: PropTypes.node
}

// Just take references from redux store and make it accessible via ReferencesContext for now
// Eventually handle retrieving and storing the references in this component w/o redux
const mapStateToProps = state => ({
  references: getReferences(state)
})

const mapDispatchToProps = {
  fetchReferences
}

export default connect(mapStateToProps, mapDispatchToProps)(ReferencesProvider)
