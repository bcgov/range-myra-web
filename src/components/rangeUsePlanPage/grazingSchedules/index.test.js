import React from 'react'
import { UserContext } from '../../../providers/UserProvider'
import { render, fireEvent } from '../../../tests/helpers/test-utils'
import GrazingSchedules from '.'
import { Formik } from 'formik'
import uuid from 'uuid-v4'
import moment from 'moment'

const plan = {
  planStartDate: moment('January 1 2021', 'MMMM D YYYY'),
  planEndDate: moment('December 31 2026', 'MMMM D YYYY'),
  agreement: {
    usage: [{ year: 2022 }, { year: 2025 }]
  },
  pastures: [
    {
      id: 1
    }
  ],
  grazingSchedules: [
    {
      id: 1,
      year: 2022,
      grazingScheduleEntries: [
        {
          pastureId: 1,
          dateIn: new Date().toISOString(),
          dateOut: new Date().toISOString(),
          graceDays: 4,
          livestockCount: 10,
          id: uuid()
        }
      ]
    },
    {
      id: uuid(),
      year: 2025,
      grazingScheduleEntries: []
    }
  ]
}

describe('Grazing Schedules', () => {
  it("doesn't render an 'Add Schedule' button if the user is a range officer", () => {
    const user = {
      roles: ['myra_range_officer']
    }

    const { queryByText } = render(
      <UserContext.Provider value={user}>
        <Formik
          initialValues={plan}
          render={({ values }) => <GrazingSchedules plan={values} />}
        />
      </UserContext.Provider>
    )

    expect(queryByText('Add Schedule')).toBeNull()
  })

  it('adds a new schedule on button press', () => {
    const { getByText } = render(
      <Formik
        initialValues={{ ...plan, grazingSchedules: [] }}
        render={({ values }) => <GrazingSchedules plan={values} />}
      />
    )

    const addButton = getByText('Add Schedule')

    fireEvent.click(addButton)

    fireEvent.click(getByText('2024'))

    expect(getByText('2024 Grazing Schedule')).toBeInTheDocument()
  })

  it('can delete schedules', () => {
    const { queryByText, getByText, getByTestId } = render(
      <Formik
        initialValues={plan}
        render={({ values }) => <GrazingSchedules plan={values} />}
      />
    )

    expect(getByText('2025 Grazing Schedule')).toBeInTheDocument()

    // The 2022 schedule should not be allowed to be deleted, since it is persisted
    expect(getByTestId('delete-button-2022')).toHaveClass('disabled')

    // Uncollapse 2025 schedule
    fireEvent.click(getByText('2025 Grazing Schedule'))

    const deleteButton = getByTestId('delete-button-2025')

    expect(deleteButton).not.toHaveClass('disabled')

    fireEvent.click(deleteButton)

    expect(getByText('Are you sure?')).toBeInTheDocument()
    fireEvent.click(getByText('OK'))

    expect(queryByText('2025 Grazing Schedule')).toBeNull()
  })

  it('can copy schedules to another year', () => {
    const { queryByText, getByText, getByTestId, queryAllByText } = render(
      <Formik
        initialValues={plan}
        render={({ values }) => <GrazingSchedules plan={values} />}
      />
    )

    expect(getByText('2022 Grazing Schedule')).toBeInTheDocument()
    expect(queryByText('2024 Grazing Schedule')).toBeNull()

    const copyButton = getByTestId('copy-button-2022')

    expect(copyButton).not.toHaveClass('disabled')

    fireEvent.click(copyButton)
    fireEvent.click(queryAllByText('2024')[1])

    expect(getByText('2024 Grazing Schedule')).toBeInTheDocument()
  })

  it('renders and matches the snapshot', () => {
    const container = render(
      <Formik
        initialValues={plan}
        render={({ values }) => <GrazingSchedules plan={values} />}
      />
    )

    const { getByText } = container

    expect(getByText('2022 Grazing Schedule')).toBeInTheDocument()
    expect(getByText('2025 Grazing Schedule')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
