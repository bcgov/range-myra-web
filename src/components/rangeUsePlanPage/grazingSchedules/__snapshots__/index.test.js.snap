// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grazing Schedules renders and matches the snapshot 1`] = `
<div
  class="rup__grazing-schedules"
>
  <div
    class="rup__content-title--editable"
  >
    <div
      class="rup__popup-header"
    >
      <div
        class="rup__content-title"
      >
        Yearly Schedules
      </div>
      <i
        aria-hidden="true"
        class="grey question small circular icon"
      />
    </div>
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="ui basic button compact labeled item pointing dropdown icon rup__grazing-schedules__add-dropdown"
      role="listbox"
      tabindex="0"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="text"
        role="alert"
      >
        Add Schedule
      </div>
      <i
        aria-hidden="true"
        class="add circle icon"
      />
      <div
        class="menu transition"
      >
        <div
          class="header"
        >
          Years
        </div>
        <div
          aria-checked="false"
          aria-selected="true"
          class="selected item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            2021
          </span>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            2023
          </span>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            2024
          </span>
        </div>
        <div
          aria-checked="false"
          aria-selected="false"
          class="item"
          role="option"
          style="pointer-events: all;"
        >
          <span
            class="text"
          >
            2026
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="rup__divider"
  />
  <ul
    class="collaspible-boxes"
  >
    <li
      class="collaspible-box"
    >
      <div
        class="collaspible-box__header"
      >
        <div
          class="collaspible-box__header__title collaspible-box__header__title--active"
        >
          <div
            class="rup__grazing-schedule__title"
          >
            <div
              style="width: 30px;"
            >
              <img
                alt="schedule icon"
                src="/images/icon_schedules.svg"
              />
            </div>
            2022
             Grazing Schedule
          </div>
          <div
            class="collaspible-box__header__right"
          >
            <div
              aria-busy="false"
              aria-disabled="false"
              aria-expanded="false"
              class="ui right pointing dropdown"
              role="listbox"
              tabindex="0"
            >
              <i
                aria-hidden="true"
                class="ellipsis vertical icon"
              />
              <div
                class="menu transition"
              >
                <div
                  aria-disabled="false"
                  aria-expanded="false"
                  class="ui left pointing dropdown link item"
                  data-testid="copy-button-2022"
                  role="listbox"
                  tabindex="0"
                >
                  <div
                    aria-atomic="true"
                    aria-live="polite"
                    class="text"
                    role="alert"
                  >
                    Copy
                  </div>
                  <i
                    aria-hidden="true"
                    class="dropdown icon"
                  />
                  <div
                    class="menu transition"
                  >
                    <div
                      class="header"
                    >
                      Years
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="true"
                      class="selected item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        2021
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        2023
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        2024
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        2026
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="item"
                  data-testid="delete-button-2022"
                  role="option"
                >
                  Delete
                </div>
              </div>
            </div>
            <i
              aria-hidden="true"
              class="chevron up icon"
              style="margin-left: 7px; margin-bottom: 3px;"
            />
          </div>
        </div>
      </div>
      <div
        class="collaspible-box__content"
      >
        <table
          class="ui unstackable table"
        >
          <thead
            class=""
          >
            <tr
              class=""
            >
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__pasture"
                >
                  Pasture
                </div>
              </th>
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__l-type"
                >
                  Livestock Type
                </div>
              </th>
              <th
                class=""
              >
                # of Animals
              </th>
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__dates"
                >
                  Date in
                </div>
              </th>
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__dates"
                >
                  Date out
                </div>
              </th>
              <th
                class=""
              >
                Days
              </th>
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__grace-days"
                >
                  Grace Days
                </div>
              </th>
              <th
                class=""
              >
                PLD
              </th>
              <th
                class=""
              >
                Crown AUMs
              </th>
              <th
                class=""
              />
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                <div
                  class="field"
                >
                  <div
                    aria-expanded="false"
                    aria-label="pasture"
                    class="ui fluid search selection dropdown"
                    id="field_dropdown_grazingSchedules.0.grazingScheduleEntries.0.pastureId"
                    name="grazingSchedules.0.grazingScheduleEntries.0.pastureId"
                    role="combobox"
                  >
                    <input
                      aria-autocomplete="list"
                      autocomplete="off"
                      class="search"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      aria-atomic="true"
                      aria-live="polite"
                      class="text"
                      role="alert"
                    />
                    <i
                      aria-hidden="true"
                      class="dropdown icon"
                    />
                    <div
                      class="menu transition"
                      role="listbox"
                    >
                      <div
                        aria-checked="true"
                        aria-selected="true"
                        class="active selected item"
                        role="option"
                        style="pointer-events: all;"
                      />
                    </div>
                  </div>
                </div>
              </td>
              <td
                class=""
              >
                <div
                  class="field"
                >
                  <div
                    aria-expanded="false"
                    aria-label="livestock type"
                    class="ui fluid search selection dropdown"
                    id="field_dropdown_grazingSchedules.0.grazingScheduleEntries.0.livestockTypeId"
                    name="grazingSchedules.0.grazingScheduleEntries.0.livestockTypeId"
                    role="combobox"
                  >
                    <input
                      aria-autocomplete="list"
                      autocomplete="off"
                      class="search"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      aria-atomic="true"
                      aria-live="polite"
                      class="text"
                      role="alert"
                    />
                    <i
                      aria-hidden="true"
                      class="dropdown icon"
                    />
                    <div
                      class="menu transition"
                      role="listbox"
                    >
                      <div
                        aria-checked="false"
                        aria-selected="true"
                        class="selected item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Cow with Calf
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Bull
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Yearling
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Horse
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Sheep
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Alpaca
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Ass
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Goat
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Llama
                        </span>
                      </div>
                      <div
                        aria-checked="false"
                        aria-selected="false"
                        class="item"
                        role="option"
                        style="pointer-events: all;"
                      >
                        <span
                          class="text"
                        >
                          Mule
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="collapsing"
              >
                <div
                  class="field"
                >
                  <div
                    class="ui fluid input"
                  >
                    <input
                      aria-label="livestock count"
                      id="field_input_grazingSchedules.0.grazingScheduleEntries.0.livestockCount"
                      name="grazingSchedules.0.grazingScheduleEntries.0.livestockCount"
                      type="text"
                      value="10"
                    />
                  </div>
                </div>
              </td>
              <td
                class="collapsing"
              >
                <div
                  class="field calendar-input--no-icon"
                >
                  <div
                    class="ui fluid icon input"
                  >
                    <i
                      aria-hidden="true"
                      class="icon"
                    />
                    <input
                      aria-label="date in"
                      name="grazingSchedules.0.grazingScheduleEntries.0.dateIn"
                      tabindex="0"
                      type="text"
                      value="Jan 26 2022"
                    />
                  </div>
                </div>
              </td>
              <td
                class="collapsing"
              >
                <div
                  class="field calendar-input--no-icon"
                >
                  <div
                    class="ui fluid icon input"
                  >
                    <i
                      aria-hidden="true"
                      class="icon"
                    />
                    <input
                      aria-label="date out"
                      name="grazingSchedules.0.grazingScheduleEntries.0.dateOut"
                      tabindex="0"
                      type="text"
                      value="Mar 9 2022"
                    />
                  </div>
                </div>
              </td>
              <td
                class="collapsing"
              >
                43
              </td>
              <td
                class="collapsing"
              >
                <div
                  class="field"
                >
                  <div
                    class="ui fluid transparent input"
                  >
                    <input
                      aria-label="grace days"
                      type="text"
                      value="4"
                    />
                  </div>
                </div>
              </td>
              <td
                class="collapsing"
              >
                0
              </td>
              <td
                class="collapsing"
              >
                0
              </td>
              <td
                class="collapsing center aligned"
              >
                <div
                  aria-expanded="false"
                  class="ui right pointing dropdown"
                  role="listbox"
                  tabindex="0"
                >
                  <i
                    aria-hidden="true"
                    class="ellipsis vertical icon"
                  />
                  <div
                    class="menu transition"
                  >
                    <div
                      aria-checked="false"
                      aria-selected="true"
                      class="selected item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        Duplicate
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        Delete
                      </span>
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </thead>
        </table>
        <div
          class="inverted-btn"
        >
          <button
            class="ui compact button"
            style="margin: 10px 0px;"
            type="button"
          >
            <i
              aria-hidden="true"
              class="add circle icon"
            />
            Add Row
          </button>
        </div>
        <div
          class="rup__grazing-schedule__AUMs"
        >
          <div
            class="rup__grazing-schedule__AUM-label"
          >
            Annual Authorized AUMs
          </div>
          <div
            class="rup__grazing-schedule__AUM-number"
          >
            0
          </div>
          <div
            class="rup__grazing-schedule__AUM-label"
          >
            Total AUMs
          </div>
          <div
            class="rup__grazing-schedule__AUM-number"
          >
            0
          </div>
        </div>
        <div
          class="rup__grazing-schedule__narrative__title"
        >
          Schedule Description
        </div>
        <div>
          <div
            class="field"
          >
            <textarea
              id="field_textarea_grazingSchedules.0.narative"
              name="grazingSchedules.0.narative"
              placeholder="Description of movement of livestock through agreement area. May include WHEN, WHERE and HOW management tools are used to create that flow. May be of particular value when an agreement consists of a single pasture or multiple unfenced pastures."
              rows="3"
              style="margin-top: 5px;"
            />
          </div>
        </div>
      </div>
    </li>
    <li
      class="collaspible-box"
    >
      <div
        class="collaspible-box__header"
      >
        <div
          class="collaspible-box__header__title"
        >
          <div
            class="rup__grazing-schedule__title"
          >
            <div
              style="width: 30px;"
            >
              <img
                alt="schedule icon"
                src="/images/icon_schedules.svg"
              />
            </div>
            2025
             Grazing Schedule
          </div>
          <div
            class="collaspible-box__header__right"
          >
            <i
              aria-hidden="true"
              class="chevron down icon"
              style="margin-left: 7px; margin-bottom: 3px;"
            />
          </div>
        </div>
      </div>
      <div
        class="collaspible-box__content collaspible-box__content--hidden"
      >
        <div
          class="ui error visible attached message"
        >
          <div
            class="content"
          >
            <i
              aria-hidden="true"
              class="warning sign icon"
              style="margin-right: 7px;"
            />
            Schedule must have at least 1 entry.
          </div>
        </div>
        <table
          class="ui unstackable bottom attached table"
        >
          <thead
            class=""
          >
            <tr
              class=""
            >
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__pasture"
                >
                  Pasture
                </div>
              </th>
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__l-type"
                >
                  Livestock Type
                </div>
              </th>
              <th
                class=""
              >
                # of Animals
              </th>
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__dates"
                >
                  Date in
                </div>
              </th>
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__dates"
                >
                  Date out
                </div>
              </th>
              <th
                class=""
              >
                Days
              </th>
              <th
                class=""
              >
                <div
                  class="rup__grazing-schedule__grace-days"
                >
                  Grace Days
                </div>
              </th>
              <th
                class=""
              >
                PLD
              </th>
              <th
                class=""
              >
                Crown AUMs
              </th>
              <th
                class=""
              />
            </tr>
          </thead>
        </table>
        <div
          class="inverted-btn"
        >
          <button
            class="ui compact button"
            style="margin: 10px 0px;"
            type="button"
          >
            <i
              aria-hidden="true"
              class="add circle icon"
            />
            Add Row
          </button>
        </div>
        <div
          class="rup__grazing-schedule__AUMs"
        >
          <div
            class="rup__grazing-schedule__AUM-label"
          >
            Annual Authorized AUMs
          </div>
          <div
            class="rup__grazing-schedule__AUM-number"
          >
            0
          </div>
          <div
            class="rup__grazing-schedule__AUM-label"
          >
            Total AUMs
          </div>
          <div
            class="rup__grazing-schedule__AUM-number"
          >
            0
          </div>
        </div>
        <div
          class="rup__grazing-schedule__narrative__title"
        >
          Schedule Description
        </div>
        <div>
          <div
            class="field"
          >
            <textarea
              id="field_textarea_grazingSchedules.1.narative"
              name="grazingSchedules.1.narative"
              placeholder="Description of movement of livestock through agreement area. May include WHEN, WHERE and HOW management tools are used to create that flow. May be of particular value when an agreement consists of a single pasture or multiple unfenced pastures."
              rows="3"
              style="margin-top: 5px;"
            />
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
